Specification/2._Basic_concepts.md

```diff
@@ -26,30 +26,30 @@ inside the same network.
 
 Message and service data structures are defined using the
 [*data structure description language* (DSDL)](/Specification/3._Data_structure_description_language).
-DSDL description is used to generate the serialization/deserialization code for a given data structure for the target
+The DSDL description is used to generate the serialization/deserialization code for a given data structure in each target
 programming language.
-The DSDL approach allows to determine the data structure size statically, thus helping to optimize the protocol
+The DSDL approach allows compilers to determine the data structure size statically, thus helping to optimize the protocol
 implementations in terms of memory consumption and performance.
 This feature is important for deeply embedded systems, where the memory footprint is critical and dynamic memory
 allocation may not be acceptable.
 
-On top of the standard data types, UAVCAN defines a set of standard high-level functions like node health monitoring,
-network discovery, time synchronization, firmware update, and more. There's a part of the specification dedicated to
+On top of the standard data types, UAVCAN defines a set of standard high-level functions including: node health monitoring,
```

Comment:
"On top of" (as in "building upon the lower level standard types") or "In addition to"?